File: src/extras-attica/libplatformdependent.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_Attica__PlatformDependent
#include <QByteArray>
#include <QIODevice>
#include <QList>
#include <QNetworkAccessManager>
#include <QNetworkReply>
#include <QNetworkRequest>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <platformdependent.h>
#include "libplatformdependent.h"
#include "libplatformdependent.hxx"

libqt_list /* of QUrl* */ Attica__PlatformDependent_GetDefaultProviderFiles(const Attica__PlatformDependent* self) {
    QList<QUrl> _ret = self->getDefaultProviderFiles();
    // Convert QList<> from C++ memory to manually-managed C memory
    QUrl** _arr = static_cast<QUrl**>(malloc(sizeof(QUrl*) * (_ret.size() + 1)));
    for (qsizetype i = 0; i < _ret.size(); ++i) {
        _arr[i] = new QUrl(_ret[i]);
    }
    libqt_list _out;
    _out.len = _ret.size();
    _out.data = static_cast<void*>(_arr);
    return _out;
}

void Attica__PlatformDependent_AddDefaultProviderFile(Attica__PlatformDependent* self, const QUrl* url) {
    self->addDefaultProviderFile(*url);
}

void Attica__PlatformDependent_RemoveDefaultProviderFile(Attica__PlatformDependent* self, const QUrl* url) {
    self->removeDefaultProviderFile(*url);
}

void Attica__PlatformDependent_EnableProvider(const Attica__PlatformDependent* self, const QUrl* baseUrl, bool enabled) {
    self->enableProvider(*baseUrl, enabled);
}

bool Attica__PlatformDependent_IsEnabled(const Attica__PlatformDependent* self, const QUrl* baseUrl) {
    return self->isEnabled(*baseUrl);
}

bool Attica__PlatformDependent_HasCredentials(const Attica__PlatformDependent* self, const QUrl* baseUrl) {
    return self->hasCredentials(*baseUrl);
}

bool Attica__PlatformDependent_LoadCredentials(Attica__PlatformDependent* self, const QUrl* baseUrl, libqt_string user, libqt_string password) {
    QString user_QString = QString::fromUtf8(user.data, user.len);
    QString password_QString = QString::fromUtf8(password.data, password.len);
    return self->loadCredentials(*baseUrl, user_QString, password_QString);
}

bool Attica__PlatformDependent_AskForCredentials(Attica__PlatformDependent* self, const QUrl* baseUrl, libqt_string user, libqt_string password) {
    QString user_QString = QString::fromUtf8(user.data, user.len);
    QString password_QString = QString::fromUtf8(password.data, password.len);
    return self->askForCredentials(*baseUrl, user_QString, password_QString);
}

bool Attica__PlatformDependent_SaveCredentials(Attica__PlatformDependent* self, const QUrl* baseUrl, const libqt_string user, const libqt_string password) {
    QString user_QString = QString::fromUtf8(user.data, user.len);
    QString password_QString = QString::fromUtf8(password.data, password.len);
    return self->saveCredentials(*baseUrl, user_QString, password_QString);
}

QNetworkReply* Attica__PlatformDependent_Get(Attica__PlatformDependent* self, const QNetworkRequest* request) {
    return self->get(*request);
}

QNetworkReply* Attica__PlatformDependent_Post(Attica__PlatformDependent* self, const QNetworkRequest* request, QIODevice* data) {
    return self->post(*request, data);
}

QNetworkReply* Attica__PlatformDependent_Post2(Attica__PlatformDependent* self, const QNetworkRequest* request, const libqt_string data) {
    QByteArray data_QByteArray(data.data, data.len);
    return self->post(*request, data_QByteArray);
}

void Attica__PlatformDependent_SetNam(Attica__PlatformDependent* self, QNetworkAccessManager* nam) {
    self->setNam(nam);
}

QNetworkAccessManager* Attica__PlatformDependent_Nam(Attica__PlatformDependent* self) {
    return self->nam();
}

void Attica__PlatformDependent_OperatorAssign(Attica__PlatformDependent* self, const Attica__PlatformDependent* param1) {
    self->operator=(*param1);
}

void Attica__PlatformDependent_Delete(Attica__PlatformDependent* self) {
    delete self;
}

File: include_original/extras-attica/libplatformdependent.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api.kde.org/attica-platformdependent.html
pub const attica__platformdependent = struct {
    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#getDefaultProviderFiles)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, allocator: std.mem.Allocator ```
    pub fn GetDefaultProviderFiles(self: ?*anyopaque, allocator: std.mem.Allocator) []QtC.QUrl {
        const _arr: qtc.libqt_list = qtc.Attica__PlatformDependent_GetDefaultProviderFiles(@ptrCast(self));
        defer qtc.libqt_free(_arr.data);
        const _ret = allocator.alloc(QtC.QUrl, _arr.len) catch @panic("attica::platformdependent.GetDefaultProviderFiles: Memory allocation failed");
        const _data: [*]QtC.QUrl = @ptrCast(@alignCast(_arr.data));
        @memcpy(_ret, _data[0.._arr.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#addDefaultProviderFile)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, url: QtC.QUrl ```
    pub fn AddDefaultProviderFile(self: ?*anyopaque, url: ?*anyopaque) void {
        qtc.Attica__PlatformDependent_AddDefaultProviderFile(@ptrCast(self), @ptrCast(url));
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#removeDefaultProviderFile)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, url: QtC.QUrl ```
    pub fn RemoveDefaultProviderFile(self: ?*anyopaque, url: ?*anyopaque) void {
        qtc.Attica__PlatformDependent_RemoveDefaultProviderFile(@ptrCast(self), @ptrCast(url));
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#enableProvider)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, baseUrl: QtC.QUrl, enabled: bool ```
    pub fn EnableProvider(self: ?*anyopaque, baseUrl: ?*anyopaque, enabled: bool) void {
        qtc.Attica__PlatformDependent_EnableProvider(@ptrCast(self), @ptrCast(baseUrl), enabled);
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#isEnabled)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, baseUrl: QtC.QUrl ```
    pub fn IsEnabled(self: ?*anyopaque, baseUrl: ?*anyopaque) bool {
        return qtc.Attica__PlatformDependent_IsEnabled(@ptrCast(self), @ptrCast(baseUrl));
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#hasCredentials)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, baseUrl: QtC.QUrl ```
    pub fn HasCredentials(self: ?*anyopaque, baseUrl: ?*anyopaque) bool {
        return qtc.Attica__PlatformDependent_HasCredentials(@ptrCast(self), @ptrCast(baseUrl));
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#loadCredentials)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, baseUrl: QtC.QUrl, user: []const u8, password: []const u8 ```
    pub fn LoadCredentials(self: ?*anyopaque, baseUrl: ?*anyopaque, user: []const u8, password: []const u8) bool {
        const user_str = qtc.libqt_string{
            .len = user.len,
            .data = user.ptr,
        };
        const password_str = qtc.libqt_string{
            .len = password.len,
            .data = password.ptr,
        };
        return qtc.Attica__PlatformDependent_LoadCredentials(@ptrCast(self), @ptrCast(baseUrl), user_str, password_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#askForCredentials)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, baseUrl: QtC.QUrl, user: []const u8, password: []const u8 ```
    pub fn AskForCredentials(self: ?*anyopaque, baseUrl: ?*anyopaque, user: []const u8, password: []const u8) bool {
        const user_str = qtc.libqt_string{
            .len = user.len,
            .data = user.ptr,
        };
        const password_str = qtc.libqt_string{
            .len = password.len,
            .data = password.ptr,
        };
        return qtc.Attica__PlatformDependent_AskForCredentials(@ptrCast(self), @ptrCast(baseUrl), user_str, password_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#saveCredentials)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, baseUrl: QtC.QUrl, user: []const u8, password: []const u8 ```
    pub fn SaveCredentials(self: ?*anyopaque, baseUrl: ?*anyopaque, user: []const u8, password: []const u8) bool {
        const user_str = qtc.libqt_string{
            .len = user.len,
            .data = user.ptr,
        };
        const password_str = qtc.libqt_string{
            .len = password.len,
            .data = password.ptr,
        };
        return qtc.Attica__PlatformDependent_SaveCredentials(@ptrCast(self), @ptrCast(baseUrl), user_str, password_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#get)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, request: QtC.QNetworkRequest ```
    pub fn Get(self: ?*anyopaque, request: ?*anyopaque) QtC.QNetworkReply {
        return qtc.Attica__PlatformDependent_Get(@ptrCast(self), @ptrCast(request));
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#post)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, request: QtC.QNetworkRequest, data: QtC.QIODevice ```
    pub fn Post(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.Attica__PlatformDependent_Post(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#post)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, request: QtC.QNetworkRequest, data: []u8 ```
    pub fn Post2(self: ?*anyopaque, request: ?*anyopaque, data: []u8) QtC.QNetworkReply {
        const data_str = qtc.libqt_string{
            .len = data.len,
            .data = data.ptr,
        };
        return qtc.Attica__PlatformDependent_Post2(@ptrCast(self), @ptrCast(request), data_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#setNam)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, nam: QtC.QNetworkAccessManager ```
    pub fn SetNam(self: ?*anyopaque, nam: ?*anyopaque) void {
        qtc.Attica__PlatformDependent_SetNam(@ptrCast(self), @ptrCast(nam));
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#nam)
    ///
    /// ``` self: QtC.Attica__PlatformDependent ```
    pub fn Nam(self: ?*anyopaque) QtC.QNetworkAccessManager {
        return qtc.Attica__PlatformDependent_Nam(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-platformdependent.html#operator-eq)
    ///
    /// ``` self: QtC.Attica__PlatformDependent, param1: QtC.Attica__PlatformDependent ```
    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.Attica__PlatformDependent_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.Attica__PlatformDependent ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.Attica__PlatformDependent_Delete(@ptrCast(self));
    }
};
